commit be17bb8fc37db4e12ea24c9480ff6442017ff656
Author: Christopher Faulet <cfaulet@haproxy.com>
Date:   Fri Sep 13 15:15:56 2019 +0200

    BUG/MEDIUM: stick-table: Properly handle "show table" with a data type argument
    
    Since the commit 1b8e68e8 ("MEDIUM: stick-table: Stop handling stick-tables as
    proxies."), the target field into the table context of the CLI applet was not
    anymore a pointer to a proxy. It was replaced by a pointer to a stktable. But,
    some parts of the code was not updated accordingly. the function
    table_prepare_data_request() still tries to cast it to a pointer to a proxy. The
    result is totally undefined. With a bit of luck, when the "show table" command
    is used with a data type, we failed to find a table and the error "Data type not
    stored in this table" is returned. But crashes may also be experienced.
    
    This patch fixes the issue #262. It must be backported to 2.0.
    
    (cherry picked from commit 4e9a83349a9b523830f79f9632ef777ab0bfcc9d)
    Signed-off-by: Christopher Faulet <cfaulet@haproxy.com>

diff --git a/src/stick_table.c b/src/stick_table.c
index 4b04f18c..8528baa0 100644
--- a/src/stick_table.c
+++ b/src/stick_table.c
@@ -3601,8 +3601,7 @@ static int table_prepare_data_request(struct appctx *appctx, char **args)
 		return 1;
 	}
 
-	if (!((struct proxy *)appctx->ctx.table.target)->table ||
-	    !((struct proxy *)appctx->ctx.table.target)->table->data_ofs[appctx->ctx.table.data_type]) {
+	if (!((struct stktable *)appctx->ctx.table.target)->data_ofs[appctx->ctx.table.data_type]) {
 		appctx->ctx.cli.severity = LOG_ERR;
 		appctx->ctx.cli.msg = "Data type not stored in this table\n";
 		appctx->st0 = CLI_ST_PRINT;
